dev/release/05-binary-upload.sh

```diff
@@ -135,3 +136,18 @@ docker_run \
     VERBOSE=${VERBOSE:-no} \
     VERSION=${version} \
     YUM_TARGETS=$(IFS=,; echo "${yum_targets[*]}")
+
+# Create a git tag to associate with a GitHub Release that will
+# have the release candidate MLTBX file as an asset. 
+# 
+# Push this tag (e.g. apache-arrow-15.0.2-rc0) to the remote
+# apache/arrow repository to trigger the GitHub Actions Workflow 
+# that creates the GitHub Release and uploads the MLTBX file.
+#
+# See .github/workflows/package.yml for details.
+if [ ${UPLOAD_MATLAB} -gt 0 ]; then
+  release_tag="apache-arrow-${version_with_rc}"
+  tag_message="Release candidate: ${version_with_rc}"
+  git tag -a ${release_tag} -m ${tag_message}
+  git push apache ${release_tag}
```

Comment:
Could you do this in `dev/release/01-prepare.sh` something like the following?

```diff
diff --git a/dev/release/01-prepare.sh b/dev/release/01-prepare.sh
index 01fa2f3d80..f5ec0a8e26 100755
--- a/dev/release/01-prepare.sh
+++ b/dev/release/01-prepare.sh
@@ -33,7 +33,7 @@ next_version=$2
 next_version_snapshot="${next_version}-SNAPSHOT"
 rc_number=$3
 
-release_tag="apache-arrow-${version}"
+rc_tag="apache-arrow-${version}-rc${rc_number}"
 release_branch="release-${version}"
 release_candidate_branch="release-${version}-rc${rc_number}"
 
@@ -45,9 +45,9 @@ release_candidate_branch="release-${version}-rc${rc_number}"
 : ${PREPARE_TAG:=${PREPARE_DEFAULT}}
 
 if [ ${PREPARE_TAG} -gt 0 ]; then
-  if [ $(git tag -l "${release_tag}") ]; then
-    echo "Delete existing git tag $release_tag"
-    git tag -d "${release_tag}"
+  if [ $(git tag -l "${rc_tag}") ]; then
+    echo "Delete existing git tag $c_tag"
+    git tag -d "${rc_tag}"
   fi
 fi
 
@@ -91,7 +91,7 @@ if [ ${PREPARE_LINUX_PACKAGES} -gt 0 ]; then
 fi
 
 if [ ${PREPARE_VERSION_PRE_TAG} -gt 0 ]; then
-  echo "Prepare release ${version} on tag ${release_tag} then reset to version ${next_version_snapshot}"
+  echo "Prepare release ${version} on tag ${rc_tag} then reset to version ${next_version_snapshot}"
 
   update_versions "${version}" "${next_version}" "release"
   git commit -m "MINOR: [Release] Update versions for ${version}"
@@ -100,5 +100,5 @@ fi
 ############################## Tag the Release ##############################
 
 if [ ${PREPARE_TAG} -gt 0 ]; then
-  git tag -a "${release_tag}" -m "[Release] Apache Arrow Release ${version}"
+  git tag -a "${rc_tag}" -m "[Release] Apache Arrow Release ${version}"
 fi
```

Reply:
> Did you mean for us to actually add the `rc_tag` in addition to the `release_tag`?

No. I mean for us to replace the `release_tag` with `rc_tag`.

> We asking because we assume we still want to create a `release_tag`.

I want to change the current workflow:

1. Create a `apache-arrow-X.Y.Z` tag for X.Y.Z RC0
2. (Found a problem for X.Y.Z RC0)
3. Remove the `apache-arrow-X.Y.Z` tag for X.Y.Z RC0
4. Create a `apache-arrow-X.Y.Z` tag for X.Y.Z RC1
5. ...

to the following:

1. Create a `apache-arrow-X.Y.Z-rc0` tag for X.Y.Z RC0
2. (Found a problem for X.Y.Z RC0)
3. Create a `apache-arrow-X.Y.Z-rc1` tag for X.Y.Z RC1
4. Vote
5. Passed
6. Create a `apache-arrow-X.Y.Z` tag from `apache-arrow-X.Y.Z-rc1` like apache/arrow-adbc and apache/arrow-flight-sql-postgresql do

Hmm. It's better that we handle the workflow change related task in a separated issue...

Reply:
Thanks for the clarification! I've created a issue #41102 to track this work. Once this PR is merged, I'll take ownership of that issue and make the required changes.

Reply:
Hi @kou,

On second thought, we think it actually makes more sense to tackle #41102 first since the new GitHub Actions Workflow (`package.yml`) requires the release candidate tag (`apache-arrow.X.Y.Z-rcN`) to be pushed. We'll put this PR on pause for a bit until that #41102 is closed.

Best,
Sarah

Reply:
It makes sense!